src/toil/batchSystems/abstractGridEngineWorker.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (C) 2015-2016 Regents of the University of California
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+import logging
+import os
+from abc import ABCMeta, abstractmethod
+from threading import Thread
+from Queue import Queue, Empty
+import time
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+class AbstractGridEngineWorker(Thread):
```

Comment:
This class should be nested in AbstractGridEngineBatchSystem and called Worker. You can then refer to it as `AbstractGridEngineBatchSystem.Worker`.  It makes the code organization is consistent between abstract and concrete classes.